### ISSUE TITLE
Roadmap 1.10: Conversational Interfaces

### ISSUE THREAD

**Issue by KnorpelSenf** [grammyjs/grammY#109]
This is a list of tasks that should be completed before the 1.10 release. All of them focus on improving how well the core library is suited to be used in [conversations](https://github.com/grammyjs/conversations).

- #100
- #103
- #105
- #127
- #131
- #157
- #176
- #188
- #247
- #201

This list is tentative. Things may be added or removed if we figure that that makes sense.

There is no ETA—it will be released once everything is done. It goes faster if you contribute.

If you want to help out but you don't know how to decide, please leave a comment here. If you want to work on some issue but you don't know how to begin, please leave a comment there. In any case, you can join [the community chat](https://t.me/grammyjs) and ask.

**Comment by KnorpelSenf** [comment]
The discussion on #136 hints at a number of new core primitives that are required to make conversational interfaces work properly. It may make sense to postpone this update, and to only include changes related to grammY conversations in that first update. I think it'll include #100, #105, and #127, plus the changes needed for conversations.

**Comment by KnorpelSenf** [comment]
The approach I mentioned above did not work out. The new approach the was started in https://github.com/grammyjs/conversations does not require any modifications of the core library. Hence, after grammY conversations are released, this update can be completed in one go.

**Comment by KnorpelSenf** [comment]
Now, a number of features look like they will drastically improve conversations, even though they may technically work without an update to the core lib. Also, there would be a lot of changes in this release.

Consequently, for those two reasons, the 1.9 release will be divided into two releases. I edited this roadmap and removed everything that will be postponed. Check out the roadmap for the next release: #202
